public/app/features/explore/Table/TableContainer.tsx

```diff
@@ -85,33 +83,45 @@ export class TableContainer extends PureComponent<Props> {
       }
     }
 
-    const mainFrame = this.getMainFrame(dataFrames);
-    const subFrames = dataFrames?.filter((df) => df.meta?.custom?.parentRowIndex !== undefined);
+    const tableData: Array<{ main: DataFrame; sub?: DataFrame[] }> = [];
+    const mainFrames = this.getMainFrames(dataFrames).filter(
+      (frame: DataFrame | undefined): frame is DataFrame => !!frame && frame.length !== 0
```

Comment:
checking for main frame length along with it existing is to be consistent with prior logic for when to display the '0 series returned' text